cgen error on use fixed array in `defer` block

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.10 b76b1ee, press to see full `v doctor` output</summary>

|V full version      |V 0.4.10 89d1aac.b76b1ee
|:-------------------|:-------------------
|OS                  |linux, Linux version 6.6.8-arch1-1 (linux@archlinux) (gcc (GCC) 13.2.1 20230801, GNU ld (GNU Binutils) 2.41.0) #1 SMP PREEMPT_DYNAMIC Thu, 21 Dec 2023 19:01:01 +0000
|Processor           |16 cpus, 64bit, little endian, 12th Gen Intel(R) Core(TM) i5-1240P
|Memory              |2.29GB/15.35GB
|                    |
|V executable        |/home/ge/.vlang/v
|V last modified time|2025-04-22 17:17:12
|                    |
|V home dir          |OK, value: /home/ge/.vlang
|VMODULES            |OK, value: /home/ge/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /tmp
|                    |
|Git version         |git version 2.49.0
|V git status        |weekly.2025.16-33-gb76b1ee7
|.git/config present |true
|                    |
|cc version          |cc (GCC) 14.2.1 20250207
|gcc version         |gcc (GCC) 14.2.1 20250207
|clang version       |clang version 19.1.7
|tcc version         |tcc version 0.9.28rc 2024-07-31 HEAD@1cee0908 (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 0134e9b9
|emcc version        |N/A
|glibc version       |ldd (GNU libc) 2.41

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  defer_fixed_arr.v && defer_fixed_arr`
```v
fn foo() {
	a := [u8(1), 2, 3]!
	defer {
		println('deffered println: ${a}')
	}
}

fn main() {
	foo()
}

```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/defer_fixed_arr.01JSFWTS3F9JBNHHK8PJFQE9RA.tmp.c:5920: error: lvalue expected
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

No C errors

The same code works fine with dynamic arrays.
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -938,7 +938,16 @@ fn (mut g Gen) assign_stmt(node_ ast.AssignStmt) {
 							|| gen_or {
 							g.expr_with_opt_or_block(val, val_type, left, var_type, is_option_auto_heap)
 						} else if val is ast.ArrayInit {
-							g.array_init(val, c_name(ident.name))
+							if val.is_fixed && ident.name in g.defer_vars {
+								g.go_before_last_stmt()
+								g.empty_line = true
+								g.write('memcpy(${c_name(ident.name)}, ')
+								g.write('(${styp})')
+								g.array_init(val, c_name(ident.name))
+								g.write(', sizeof(${styp}))')
+							} else {
+								g.array_init(val, c_name(ident.name))
+							}
 						} else if val_type.has_flag(.shared_f) {
 							g.expr_with_cast(val, val_type, var_type)
 						} else if val in [ast.MatchExpr, ast.IfExpr]

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -938,15 +938,16 @@ fn (mut g Gen) assign_stmt(node_ ast.AssignStmt) {
 							|| gen_or {
 							g.expr_with_opt_or_block(val, val_type, left, var_type, is_option_auto_heap)
 						} else if val is ast.ArrayInit {
+							cvar_name := c_name(ident.name)
 							if val.is_fixed && ident.name in g.defer_vars {
 								g.go_before_last_stmt()
 								g.empty_line = true
-								g.write('memcpy(${c_name(ident.name)}, ')
+								g.write('memcpy(${cvar_name}, ')
 								g.write('(${styp})')
-								g.array_init(val, c_name(ident.name))
+								g.array_init(val, cvar_name)
 								g.write(', sizeof(${styp}))')
 							} else {
-								g.array_init(val, c_name(ident.name))
+								g.array_init(val, cvar_name)
 							}
 						} else if val_type.has_flag(.shared_f) {
 							g.expr_with_cast(val, val_type, var_type)

--- a/vlib/v/tests/defer/defer_fixed_array_test.v
+++ b/vlib/v/tests/defer/defer_fixed_array_test.v
@@ -1,0 +1,11 @@
+fn foo() {
+	a := [u8(1), 2, 3]!
+	defer {
+		println('deffered println: ${a}')
+		assert a == [u8(1), 2, 3]!
+	}
+}
+
+fn test_main() {
+	foo()
+}

